backend/models/ewf_gorm_store.go

```diff
@@ -0,0 +1,85 @@
+package models
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/xmonader/ewf"
+	"gorm.io/gorm"
+)
+
+type EWFGormStore struct {
+	db *gorm.DB
+}
+
+type GormWorkflowStruct struct {
+	UUID   string `gorm:"primaryKey;column:uuid"`
+	Name   string `gorm:"column:name;not null;index"`
+	Status string `gorm:"column:status;not null;index"`
+	Data   []byte `gorm:"column:data;not null"`
+}
+
+func NewGormStore(db *gorm.DB) *EWFGormStore {
+	return &EWFGormStore{db: db}
+}
+
+func (s *EWFGormStore) Setup() error {
+	return s.db.AutoMigrate(&GormWorkflowStruct{})
+}
+
+func (s *EWFGormStore) SaveWorkflow(ctx context.Context, workflow *ewf.Workflow) error {
+	data, err := json.Marshal(workflow)
+	if err != nil {
+		return fmt.Errorf("failed to marshal workflow: %w", err)
+	}
+
+	gormWorkflow := GormWorkflowStruct{
+		UUID:   workflow.UUID,
+		Name:   workflow.Name,
+		Status: string(workflow.Status),
+		Data:   data,
+	}
+
+	return s.db.WithContext(ctx).Save(gormWorkflow).Error
```

Comment:
this will always overwrite the record?

Reply:
Yes, ewf is using this to update the workflow in the store on every step it passes, failure, or completion of the workflow